### PR TITLE
[MIG][13.0] crm_project: From Odoo core to OCA

### PR DESCRIPTION
**Pull request by pedrobaeza**
This module has been removed from Odoo in favor of `crm_helpdesk` in enterprise, but this still makes sense for community edition where no helpdesk exists or for enterprise that don't want to install an extra app (helpdesk).

* Initial commit with 12.0 Odoo CE code.
* Isolated black, prettier and isort commit.
* Then standard procedure to migrate.
* Adapted view. Now it's seen in both leads and opportunities.
* Added tests.
* Adjusted security to project users.

@Tecnativa TT19727

### CODE REVIEW

File: crm_project/README.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+#. You need to have at least "Project User" and "Salesman" permissions to be
+   able to convert leads to tasks.
+
+Usage
+=====
+
+#. Go to *CRM > Sales > My pipeline*.
+#. Open one of the existing leads or create one.
+#. Click on the button "Create Task".
```
Comment (joao-p-marques) [inline crm_project/README.rst]:
```suggestion
#. Click on the button "Convert to Task".
```

**Review by joao-p-marques — APPROVED** [review]
Code and functional review :+1:

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍

**Pull request merged**